spynnaker/pyNN/models/common/recordable.py

```diff
@@ -0,0 +1,120 @@
+from six import add_metaclass
+
+from spinn_utilities.abstract_base import AbstractBase, abstractmethod
+
+abstractstaticmethod = abstractmethod
+
+
+@add_metaclass(AbstractBase)
+class Recordable(object):
+    """ Indicates that spikes can be recorded from this object
+    """
+
+    __slots__ = ()
+
+    # Here we have to choose between using @abstractstatic or @staticmethod
+    # Using both either fails or cause @abstractstatic to be ignored
+    # However even abstractstatic only enforces that one of the
+    # abstract_x_recordable interfaces implements the method not all.
```

Comment:
This is definitely an issue - I don't think python provides any way around this (Java doesn't know how to do this either ;) )